docs/refguide/computed.md

```diff
@@ -0,0 +1,222 @@
+For example, if we'd extend the above example with calling `console.log(order.total)` twice, after we called `stop()`, the value would be recomputed twice.
+
+This allows MobX to automatically suspend computations that are not actively in use,
+to avoid unnecessary updates to computed values that are not being accessed. But if a computed property is _not_ in use by some reaction, computed expressions are evaluated each time their value is requested, so they just behave like a normal property.
+
+So if you fiddle around, computed properties might not seem efficient. But when applied in a project that uses `observer`, `autorun` etc, they become very efficient.
+
+The following code demonstrates the issue.
+
+```javascript
+// OrderLine has a computed property `total`
+const line = new OrderLine(2.0)
+
+// if you access line.total outside of a reaction it is recomputed every time
+setInterval(() => {
+    console.log(line.total)
+}, 60)
+```
+
+It can be overridden by setting annotating with the `keepAlive` flag ([try it](https://codesandbox.io/s/computed-3cjo9?file=/src/index.tsx)) or by creating a no-op `autorun(() => { someObject.someComputed })` (which can nicely be cleaned up later if needed).
+Note that both solutions have the risk of creating memory leaks; changing the default behavior here is an anti-pattern.
+
+MobX can be configured to report an error when computeds are accessed outside an reactive context with the [`computedRequiresReaction`](configure#computedrequiresreaction) option.
+
+</details>
+
+<details id="computed-setter"><summary>Tip: computed values can have setters<a href="#computed-setter" class="tip-anchor"></a></summary>
+
+It is possible to define a [setter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/set) for computed values as well. Note that these setters cannot be used to alter the value of the computed property directly,
+but they can be used as 'inverse' of the derivation. Setters are automatically marked as actions. For example:
+
+```javascript
+class Dimension {
+    length = 2
+
+    constructor() {
+        makeAutoObservable(this)
+    }
+
+    get squared() {
+        return this.length * this.length
+    }
+    set squared(value) {
+        this.length = Math.sqrt(value)
+    }
+}
+```
+
+</details>
+
+<details id="computed-struct"><summary>🚀 Tip: `computed.struct` for comparing output structurally <a href="#computed-struct" class="tip-anchor"></a></summary>
+
+If the output of a computed value, that is structurally equivalent to the previous computation, doesn't need to notify observers, `computed.struct` can be used. It will make a structural comparison first (rather than a reference equality check) before notifying observers. Example:
```

Comment:
```suggestion
If the output of a computed value that is structurally equivalent to the previous computation doesn't need to notify observers, `computed.struct` can be used. It will make a structural comparison first (rather than a reference equality check) before notifying observers. Example:
```